### ISSUE TITLE
"An unknown error has occurred" when creating test account

### ISSUE THREAD

**Issue by bswinnerton** [arsduo/koala#462]
When trying to create a Facebook test user:

``` ruby
test_users_client = Koala::Facebook::TestUsers.new(app_id: ENV.fetch('FACEBOOK_APP_ID'), secret: ENV.fetch('FACEBOOK_APP_SECRET'))
test_users_client.create(true, 'user_about_me,user_friends,user_work_history')
```

I receive the following error:

```
Koala::Facebook::ServerError: type: OAuthException, code: 1, message: An unknown error has occurred. [HTTP 500]
    from ./vendor/bundle/gems/koala-1.11.1/lib/koala/api.rb:73:in `api'
```

It does not appear to be an issue with the `APP_ID` or `APP_SECRET`, because if I change either of those, I receive a different error:

```
Koala::Facebook::OAuthTokenRequestError: type: OAuthException, code: 1, message: Error validating client secret. [HTTP 400]
    from .vendor/bundle/gems/koala-1.11.1/lib/koala/oauth.rb:368:in `fetch_token_string'
```

Before even trying to execute `.create`.

For what it's worth, I receive the same error when trying:

``` ruby
test_users_client.create(false)
```

**Comment by Dgrafmyre** [comment]
I have been having the same issue since last night with the above error. However, Today I am now getting this error:

```@test_users = Koala.Facebook.TestUsers.new(app_id: Figaro.env.facebook_app_id, secret: Figaro.env.facebook_app_secret)
@test_users.create_network(10, true, 'public_profile, email, user_friends')```

```Koala::Facebook::ClientError: type: GraphMethodException, code: 100, message: API calls from the server require an appsecret_proof argument [HTTP 400]
from /Users/dgrafmyre/.rvm/gems/ruby-2.2.1@shakabnb/gems/koala-2.0.0/lib/koala/api/graph_api.rb:500:in `block in graph_call'```

```@test_users.create(true, 'public_profile, email, user_friends')```

Returns a new test user. I don't know if Facebook migrating off API1v is messing with things or maybe something else has changed.

I gave up creating a new test user network, and went to add them in the developer console. Low and behold I had a ton of new test_users created?

Also, I have my app set to "Required App Secret".

**Comment by arsduo** [comment]
Hey all,

Are you still having these issues? Sorry about the late response.

@dylangrafmyre, it looks like your problem was related to the appsecret_proof — when you sent that, were you still seeing the problem?

One suggestion for any Facebook issues: turn on Koala's debug mode, then try the same command in Facebook's developer console online. That gives you the ability to see if it's a Koala issue or a Facebook one.

Alex

**Comment by samnang** [comment]
@arsduo I got the same error when I tried to make two users to be friends.

```
Failure/Error: test_users.befriend(fb_user_a, fb_user_b)
Koala::Facebook::ClientError:
  type: GraphMethodException, code: 100, message: API calls from the server require an appsecret_proof argument [HTTP 400]
```

It works just fine if I can these lines to be https://github.com/arsduo/koala/blob/ffa1dc61cf06a5c875567c1442db04956754b811/lib/koala/test_users.rb#L149-L150

```ruby
u1_graph_api = API.new(user1_token, @secret)
u2_graph_api = API.new(user2_token, @secret)
```
So it seems facebook needs appsecret_proof for `befriends` api.

@dylangrafmyre have you found any solution?

**Comment by arsduo** [comment]
Ah, that makes sense. There's a PR open now to add the secret (#542), so I'll resolve this -- once that's merged in and 3.0.0 is released, please give it a try (if you're still using these APIs, of course -- it's been a while).
